test/Tests/src/Data/Time/Date_Spec.enso

```diff
@@ -139,6 +139,22 @@ spec_with name create_new_date parse_date =
             date_1>=datetime . should_fail_with Type_Error_Data
             date_1==datetime . should_be_false
 
+        Test.specify "should create date before epoch start" <|
+            (create_new_date 100 1 2).year   . should_equal 100
+            (create_new_date 100 1 2).month  . should_equal 1
+            (create_new_date 100 1 2).day    . should_equal 2
+
+        Test.specify "Gregorian calendar related functionality should fail before epoch start" <|
+            handler err = err . should_equal Time_Error.epoch_start
+            dates_before_epoch = [(create_new_date 100), (create_new_date 500 6 3)]
+            dates_before_epoch.each date->
+                Panic.catch Time_Error_Data date.week_of_year handler
+                Panic.catch Time_Error_Data date.is_leap_year handler
+                Panic.catch Time_Error_Data date.day_of_week handler
+                Panic.catch Time_Error_Data date.length_of_month handler
+                Panic.catch Time_Error_Data date.length_of_year handler
+                Panic.catch Time_Error_Data date.week_of_year handler
```

Comment:
That's not how we test panics. If the function does not panic, it will do nothing and the test will succeed.

For panics we have `Test.expect_panic_with`. But indeed your functions do not panic - you should use `result . should_fail_with Error_Ctor` instead.